### PR TITLE
feature/fetch-data: add NotesDataAPI class that handles fetching and storing the data

### PR DESCRIPTION
**Pull request by CozmaRares**

### CODE REVIEW

File: src/fetch-data.js [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+}
+
+// TODO: rethink the functions below after we have the markup for a note
+
+/**
+ * Generate the HTML code for a note with details
+ * @param {NoteWithDescription} note
+ * @returns {Node}
+ */
+function NoteWithDescriptionHTML(note) {
```
Comment (popsilviu) [inline src/fetch-data.js:92]: Look into using https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template, I may or may not remember if this is the correct usage for reusable HTML.

Comment (popsilviu) [reply src/fetch-data.js:92]: As a not this was nice of you to add but a bit beyond the scope of the ticket.

File: src/fetch-data.js [hunk 2]
```diff
@@ -0,0 +1,197 @@
+// @ts-check
```
Comment (popsilviu) [inline src/fetch-data.js:1]: Please retain the JSdoc but loose the type checking, we don't need that for now and is not enforceable as we don't have a build pipeline for our project

**Review by popsilviu — CHANGES_REQUESTED** [review]

**Comment by CozmaRares** [comment]
added implementation for the tickets:
      - transmit create data to api
      - transmit update data to api
      - transmit delete data to api

**Review by popsilviu — APPROVED** [review]

**Pull request merged**